Update `PlacementRule` --> `Placement` for GRC

dhaiducek opened the pull request:
(Setting to "Draft" until `2.10_stage` is open and it can switch over.)

ref: https://issues.redhat.com/browse/ACM-9571

dhaiducek committed:
Update `PlacementRule` --> `Placement` for GRC

ref: https://issues.redhat.com/browse/ACM-9571
Co-authored-by: Mikela Jackson <mdockery@redhat.com>
Signed-off-by: Dale Haiducek <19750917+dhaiducek@users.noreply.github.com>

dockerymick commented:
@dhaiducek thanks for creating the PR, I updated the branch to 2.10 and will review it shortly. Thanks!

dockerymick commented on governance/create_policy.adoc:
```diff
@@ -17,7 +17,9 @@ You can create a security policy from the command line interface (CLI) or from t
-*Important:* You must define a placement and placement binding to apply your policy to a specific cluster. Enter a valid value for the _Cluster selector_ field to define a `Placement` and `PlacementBinding`.
+*Important:*
+* You must define a placement and placement binding to apply your policy to a specific cluster. Enter a valid value for the cluster _Label selector_ field to define a `Placement` and `PlacementBinding`.
+* In order to use `Placement`, a `ManagedClusterSet` must be bound to the `Placement` namespace with a `ManagedClusterSetBinding`. Refer to xref:../cluster_lifecycle/create_clustersetbinding.adoc#creating-managedclustersetbinding[Creating a _ManagedClusterSetBinding_ resource] for additional details.
```
```suggestion
* In order to use `Placement` resource, a `ManagedClusterSet` resource must be bound to the namespace of the `Placement` resource with a `ManagedClusterSetBinding` resource. Refer to xref:../cluster_lifecycle/create_clustersetbinding.adoc#creating-managedclustersetbinding[Creating a _ManagedClusterSetBinding_ resource] for additional details.
```

dockerymick reviewed (commented):
Everything seems pretty good to me! Thanks Dale! Quick question, is PlaceentRule officially removed for 2.10?

dhaiducek commented:
> Everything seems pretty good to me! Thanks Dale! Quick question, is PlacementRule officially removed for 2.10?

It's not removed that I'm aware of (that'd be a question for App Lifecycle--I think it might be removed in 2.11?), but we wanted to be prepared well before it was removed.

dockerymick commented on governance/create_policy.adoc:
```diff
@@ -17,7 +17,9 @@ You can create a security policy from the command line interface (CLI) or from t
-*Important:* You must define a placement and placement binding to apply your policy to a specific cluster. Enter a valid value for the _Cluster selector_ field to define a `Placement` and `PlacementBinding`.
+*Important:*
+* You must define a placement and placement binding to apply your policy to a specific cluster. Enter a valid value for the cluster _Label selector_ field to define a `Placement` and `PlacementBinding`.
+* In order to use a `Placement` resource, a `ManagedClusterSet` resource must be bound to the namespace of the `Placement` resource with a `ManagedClusterSetBinding` resource. Refer to xref:../cluster_lifecycle/create_clustersetbinding.adoc#creating-managedclustersetbinding[Creating a _ManagedClusterSetBinding_ resource] for additional details.
```
```suggestion
* In order to use a `Placement` resource, a `ManagedClusterSet` resource must be bound to the namespace of the `Placement` resource with a `ManagedClusterSetBinding` resource. Refer to link:../cluster_lifecycle/create_clustersetbinding.adoc#creating-managedclustersetbinding[Creating a _ManagedClusterSetBinding_ resource] for additional details.
```

dockerymick commented on governance/create_policy.adoc:
```diff
@@ -17,13 +17,9 @@ You can create a security policy from the command line interface (CLI) or from t
-View more descriptions of the policy YAML files in the xref:../governance/policy_overview.adoc#policy-overview[Policy overview]. 
+*Important:*
+* You must define a placement and placement binding to apply your policy to a specific cluster. The `PlacementBinding` resource binds the placement. Enter a valid value for the cluster _Label selector_ field to define a `Placement` and `PlacementBinding` resource.
+* In order to use a `Placement` resource, a `ManagedClusterSet` resource must be bound to the namespace of the `Placement` resource with a `ManagedClusterSetBinding` resource. Refer to xref:../clusters/cluster_lifecycle/create_clustersetbinding.adoc#creating-managedclustersetbinding[Creating a _ManagedClusterSetBinding_ resource] for additional details.  
```
```suggestion
* In order to use a `Placement` resource, a `ManagedClusterSet` resource must be bound to the namespace of the `Placement` resource with a `ManagedClusterSetBinding` resource. Refer to link:../clusters/cluster_lifecycle/create_clustersetbinding.adoc#creating-managedclustersetbinding[Creating a _ManagedClusterSetBinding_ resource] for additional details.  
```

dockerymick commented on governance/policy_set_ctrl.adoc:
```diff
@@ -7,7 +7,9 @@ Additionally, when a policy is a part of multiple policy sets, existing and new
 
-*Note:* The {product-title-short} hardening sample policy set uses cluster placement. If you use cluster placement, bind the namespace containing the policy to the managed cluster set. See link:../gitops/deploy_gitops.adoc#deploying-policies-to-your-cluster[Deploying policies to your cluster] for more details on using cluster placement.
+*Note:*
+* The {product-title-short} hardening sample policy set uses cluster placement. If you use cluster placement, bind the namespace containing the policy to the managed cluster set. See link:../gitops/deploy_gitops.adoc#deploying-policies-to-your-cluster[Deploying policies to your cluster] for more details on using cluster placement.
```
```suggestion
* The {product-title-short} sample policy set uses cluster placement. If you use cluster placement, bind the namespace containing the policy to the managed cluster set. See link:../gitops/deploy_gitops.adoc#deploying-policies-to-your-cluster[Deploying policies to your cluster] for more details on using cluster placement.
```

dockerymick commented on governance/policy_set_ctrl.adoc:
```diff
@@ -7,7 +7,10 @@ Additionally, when a policy is a part of multiple policy sets, existing and new
 
 *Deprecated:* `PlacementRule`
```
I noticed that you removed this in other files, I assume that we also want to remove this instance as well?

dhaiducek replied:
Good catch, yes we do!

jc-berger reviewed (approved):
/lgtm

dhaiducek commented:
Thanks for the reviews! It looked like the `../clusters/cluster_lifecycle/` paths are now `../cluster_lifecycle`, so I updated the rest of them to the new path. Also, I see you were updating `xref` to `link`, so I updated some others contained in this PR.

dockerymick commented:
@jc-berger thanks for your review and @dhaiducek thanks for accepting my suggestions and changes. Approving and merging this PR
